src/Command/CommandDispatcherInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace SystemCtl\Command;
+
+use SystemCtl\Exception\CommandFailedException;
+
+interface CommandDispatcherInterface
+{
+    /**
+     * Timeout after which the dispatcher failes the execution
+     *
+     * @param int $timeout
+     *
+     * @return CommandDispatcherInterface
+     */
+    public function setTimeout(int $timeout): CommandDispatcherInterface;
+
+    /**
+     * Set basic binary to dispatch
+     *
+     * @param string $binary
+     *
+     * @return CommandDispatcherInterface
+     */
+    public function setBinary(string $binary): CommandDispatcherInterface;
+
+    /**
+     * Dispatch given commands against implementers logic and creating a new command
+     * to read results
+     *
+     * @param array $commands
+     *
+     * @return CommandInterface
+     * @throws CommandFailedException
+     */
+    public function dispatch(...$commands): CommandInterface;
```

Comment:
No because of the argument unpacking.
Since `dispatch` is a variadic function and accepts any number of arguments.
```
->dispatch('a', 'b'); // ['a', 'b']
```

So does argument unpacking the same thing reverse. If you take an array and pass it with `...$array`. Every element will be passed into the funktion as a single argument.

Yes this seems like a stupid way for the `listUnits` function. Since i unpack the array into single argument and then using the variadic combine them back into an array. But tbh there is not much difference here. This was simply an choice to choose vardiadics instead of one single parameter as `array`.